catalog_tools/analysis/estimate_beta.py

```diff
@@ -53,7 +53,41 @@ def estimate_beta_utsu(magnitudes: np.ndarray, mc: float, delta_m: float = 0
     return beta
 
 
-def estimate_beta_elst(magnitudes: np.ndarray) -> float:
+def differences(magnitudes: np.ndarray) -> np.ndarray:
+    """returns all the differences between the magnitudes.
+
+    Args:
+        magnitudes: vector of magnitudes differences, sorted in time (first
+                    entry is the earliest earthquake)
+
+    Returns: array of positive differences of the input
+    """
+    mag_diffs = np.array([])
+    for ii, mag in enumerate(magnitudes):
+        loop_mag = np.delete(magnitudes, [ii], axis=0)
+        mag_diffs = np.append(mag_diffs, loop_mag - mag)
+    return mag_diffs
+
+
+def differences_successive(magnitudes: np.ndarray) -> np.ndarray:
+    """returns the differences of successive magnitudes.
+
+    Args:
+        magnitudes: vector of magnitudes, sorted in time (first
+                    entry is the earliest earthquake)
+
+    Returns: array of positive differences of the input
+    """
+    temp_mags1 = np.append([0], magnitudes)
+    temp_mags2 = np.append(magnitudes, [0])
+    mag_diffs = temp_mags1 - temp_mags2
```

Comment:
I guess it is a question of definition, but wouldn't it make more sense to use the differences M2 - M1 (if M1 occurs first and M2 second)?
also, np.diff(magnitudes) does exactly this, no need to implement it. if you want to keep your version of M1-M2, then it would be -np.diff(magnitudes)